### PR TITLE
Introduce Agent at the CLI composition root (Fixes #2200)

### LINKED ISSUE #2200: Introduce Agent at the CLI composition root
## Parent

Subissue of #1595.

## Architectural intent

llxprt already has reusable runtime/API work in core and subpackages. This subissue is part of #1595 deficiency-closure and migration work: make that public Agent/runtime API complete and clean enough that clients consume the shared runtime instead of reconstructing providers, tools, auth, settings, MCP, policy, hooks, history, and the chat loop from internals.

Primary near-term clients are the interactive CLI and non-interactive prompt mode. A2A and future integrations should eventually be able to use the same API, but A2A is intentionally not a major focus for this release because it was ported from upstream incompletely and needs holistic follow-up work. Do not expand this subissue into a full A2A repair; just avoid API or boundary choices that would make future A2A adoption harder.

## Goal

Introduce the Agent at the CLI composition root and pass a single Agent instance into the interactive UI while preserving existing Config/bootstrap behavior during migration.

## Scope

- Build Config as today, then create/adopt an Agent using the supported public Agent entrypoint, expected to be fromConfig for this staged migration.
- Thread the Agent through startInteractiveUI, AppWrapper, AppContainer, and AppContainerRuntime.
- Keep Config available only as a temporary migration bridge where unavoidable, with every remaining Config use documented for later removal.
- Own Agent lifecycle explicitly, including dispose behavior, shutdown ordering, cancellation, and preserving caller-owned Config semantics.
- Begin shrinking cli.tsx and cliBootstrap.tsx by moving wiring into focused composition helpers.
- Keep this focused on the interactive CLI composition root; do not pull in A2A remediation.

## Acceptance criteria

- Interactive bootstrap creates exactly one Agent for the UI path.
- AppContainer receives an Agent instance.
- Existing CLI startup/auth/provider behavior remains user-visible compatible.
- Shutdown disposes the Agent without prematurely tearing down caller-owned Config resources.
- Any remaining Config bridge use is explicit migration debt for later subissues.
- New tests protect the composition root and lifecycle.

## Suggested tests

- CLI bootstrap test proving fromConfig is called once with the existing Config.
- AppContainer render/compile test using a fake minimal Agent.
- Agent lifecycle/dispose test covering normal exit and interrupted startup.
- Regression test that the legacy runtime is not constructed twice.

### PR DESCRIPTION
**Pull request by acoliver**
## Summary

Introduces the public **Agent** at the interactive CLI composition root, as the first staged step of the larger CLI-to-core-API migration (parent #1595). The CLI now builds `Config` exactly as before, then creates **exactly one** foreground `Agent` by adopting that `Config` (and the existing session `MessageBus`) through the supported `fromConfig` entrypoint. That single `Agent` instance is threaded through the interactive UI component chain and its lifecycle is owned via the existing cleanup machinery.

This stage is intentionally a **bridge**: streaming hooks are left behaviorally unchanged, and `Config` remains available as documented migration debt for later subissues.

Fixes #2200.

### New composition helper

- `packages/cli/src/cliAgentBootstrap.ts` — exports `createForegroundAgent({ config, sessionMessageBus })`, the single creation point for the interactive Agent. It calls `fromConfig({ config, messageBus: sessionMessageBus })` so no second `ProviderManager`/`MessageBus` is constructed, keeps `configOwnership` caller-owned (the `fromConfig` default), and registers the Agent's `dispose()` via `registerCleanup` so shutdown disposes the Agent on both normal and interrupted/fatal startup exits.

### Wiring into the interactive dispatch path (`cli.tsx`)

- The interactive branch of `dispatchInteractiveOrNonInteractive` now calls `createForegroundAgent` exactly once before invoking `startInteractiveUI`.
- The explicit interactive `triggerSessionStartHook(config, SessionStartSource.Startup)` call is removed, because `fromConfig` fires `SessionStart` internally through the same core hook (verified: `finalizeAgent` -> `assembleFacade` -> `agent.hooks.triggerSessionStart()` -> `triggerSessionStartHook(config, SessionStartSource.Startup)`). This preserves the existing single-fire behavior.
- The **non-interactive** branch is untouched: it constructs no Agent and keeps its own explicit `triggerSessionStartHook` call.

### Threading the Agent through the UI

- `startInteractiveUI` takes a new `agent: Agent` parameter and passes it as a required `agent` prop to `AppWrapper`.
- `agent` is added as a required field to `AppProps` (`App.tsx`), `AppContainerProps` (`AppContainer.tsx`), `AppContainerRuntimeProps` (`AppContainerRuntime.tsx`), and `useAppBootstrap` (props + result), forwarded through `AppWrapper -> AppWithState -> AppContainer -> AppContainerRuntime -> useAppBootstrap`. Because the prop is required, the type checker enforces end-to-end delivery.
- The remaining `config.getAgentClient()` consumption point in `useAppInput.ts` is annotated as a migration bridge referencing #1595; streaming behavior is unchanged in this stage.

## Lifecycle / ownership semantics (verified against the agents package)

- `fromConfig` **adopts** the caller-owned `Config` and `ProviderManager` — it does not reconstruct the runtime.
- Because ownership is caller-owned, the Agent's `dispose()` aborts active runs and fires `SessionEnd`, but deliberately **skips** `config.dispose()`. Existing recording/Config teardown remains owned by the existing bootstrap.
- `dispose()` is idempotent and runs through `registerCleanup`/`runExitCleanup` on every exit path.

## Tests

New behavioral tests (no mock theater — only the `fromConfig` boundary is mocked; the real cleanup machinery is exercised):

- `packages/cli/src/cliAgentBootstrap.test.ts`:
  - `fromConfig` is called exactly once with the existing `Config` and session `MessageBus` instances (identity equality, proving adoption rather than reconstruction).
  - The created Agent is returned.
  - A cleanup is registered that disposes the Agent on normal exit (alongside a later-registered interactive-UI cleanup, asserting both run).
  - The Agent is still disposed when startup is interrupted before the UI registers cleanup.
  - The caller-owned `Config` is **not** torn down when the Agent is disposed.
  - When `fromConfig` rejects, no disposal cleanup is registered.

Existing tests updated to pass the required `agent` prop / parameter: `cli.test.tsx` (asserts `createForegroundAgent` is called once on the interactive path), `cli.startInteractiveUI.test.tsx`, `cli.renderOptions.test.tsx`, `cli.provider-init.test.ts`, and the `AppContainer.*` mount/keybindings/render-budget tests.

## Acceptance criteria

- [x] Interactive bootstrap creates exactly one Agent for the UI path.
- [x] AppContainer receives an Agent instance.
- [x] Existing CLI startup/auth/provider behavior remains user-visible compatible (single `SessionStart` fire preserved; non-interactive path untouched).
- [x] Shutdown disposes the Agent without prematurely tearing down caller-owned Config resources.
- [x] Any remaining Config bridge use is documented as explicit migration debt for later subissues (#1595).
- [x] New tests protect the composition root and lifecycle.

## Verification

- `npm run build` — pass
- `npm run typecheck` — pass
- `npm run lint` + `npm run lint:eslint-guard` — pass
- `npm run format` — pass
- CLI test suite — 410 files / 4862 passed / 6 skipped
- Smoke: `node scripts/start.js --profile-load ollamakimi "write me a haiku and nothing else"` — produced a valid haiku
- Open Code Review (ocr) — run; both findings addressed (lifecycle tests differentiated + fromConfig-rejection edge case added)

## Out of scope (intentional)

- No rewrite of streaming hooks; `config.getAgentClient()` consumers are migrated in later subissues under #1595.
- No A2A remediation.

### CODE REVIEW

File: packages/cli/src/ui/__tests__/AppContainer.keybindings.test.tsx [hunk 1]
```diff
@@ -19,9 +19,17 @@ import {
...
 import type { Config } from '@vybestack/llxprt-code-core';
+import type { Agent } from '@vybestack/llxprt-code-agents';
 import { Command } from '../keyMatchers.js';
 import { defaultKeyBindings } from '../../config/keyBindings.js';
 
+function createMockAgent(config: Config): Agent {
+  return {
+    dispose: vi.fn().mockResolvedValue(undefined),
+    getConfig: () => config,
+  } as unknown as Agent;
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/ui/__tests__/AppContainer.keybindings.test.tsx]: Done in 8ae807410. Extracted the duplicated createMockAgent stub into a single shared helper at packages/cli/src/test-utils/mockAgent.ts and imported it in the keybindings, mount, and render-budget suites, deleting all three local copies. Future Agent-prop contract changes during the #1595 migration are now a one-file edit.

File: packages/cli/src/cliAgentBootstrap.test.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+  it('does not tear down the caller-owned config when disposing the agent', async () => {
+    const dispose = vi.fn();
+    config = makeConfig({ dispose });
+    fakeAgent.getConfig = () => config;
+
+    await createForegroundAgent({ config, sessionMessageBus });
+    await runExitCleanup();
+
+    expect(dispose).not.toHaveBeenCalled();
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/cliAgentBootstrap.test.ts]: Done in 8ae807410. Removed this test: with fromConfig mocked it only exercised an inert stub (mock theater). The caller-owned Config disposal contract is owned and verified against the real fromConfig/dispose flow in the agents package (packages/agents/src/api/__tests__/fromConfig.behavior.test.ts, T7/T7b). Left an inline comment in cliAgentBootstrap.test.ts pointing to that authoritative coverage.
